### PR TITLE
Updated for new LK

### PR DESCRIPTION
**Pull request by bvp**

### CODE REVIEW

**Comment by kovetskiy** [comment]
Hello there, thank you for such a big contribution to the project.

Unfortunately, I don't use Yota anymore, and I can suggest two ways depending on what you want from this PR:
1. I can merge this PR as is if you just want the code changes to be in the upstream
2. I can provide a comprehensive code review before merging which might require some rewrites from your side

Just let me know what you expect and I'll give it to you :)

**Comment by bvp** [comment]
>     2. I can provide a comprehensive code review before merging which might require some rewrites from your side

I choose the blue pill

File: main.go
Comment (kovetskiy) [inline main.go:64]: I don't think it's still viable to use this InsecureSkipVerify as true. Have no idea why it was true, but we really should get rid of it since it's insecure.

File: main.go
Comment (kovetskiy) [inline main.go:67]: Is there any motivation behind this particular timeout? Why not default one?

I mean I understand that it's like "a minute is enough for everyone" but it's more a code review-ish thing to ask since it's a question you might have when you look into a code next year and start thinking about what could be the reason of such a timeout "was it solving some specific issue? if so, then which one?". I suggest adding a comment here why you picked a minute or just use the default one.

File: main.go
Comment (kovetskiy) [inline main.go:87]: Oof, I don't think we can do anything if the method Do() has failed and should really interrupt the execution and return the error if the request was failed. It means you will also have to change the method signature to be `(DevicesInfo, error)`.

If the error is not nil the `res` will also be `nil` (since it's the way most of Go packages/funcs work) which means you will actually get a panic in the next lines where you work with `res` if the request was failed.

File: main.go
Comment (kovetskiy) [inline main.go:91]: Nope, you can't ignore errors. Just catch them, handle them, and give them to the user.

File: main.go
Comment (kovetskiy) [inline main.go:93]: You forgot to check the error here. But I actually think it's something from your debugs where you wanted to check what is coming in the response. Looks like you just forgot to remove these lines.

File: main.go
Comment (kovetskiy) [inline main.go:98]: Same thing with log, we should return the error.

File: main.go
Comment (kovetskiy) [inline main.go:112]: No need to defer it since you can just return the value like `return res.Body.Close()`.

File: main.go
Comment (kovetskiy) [inline main.go:104]: Naming the return variable is redundant here.

File: main.go
Comment (kovetskiy) [inline main.go:136]: Same thing with error handling here.

File: main.go
Comment (kovetskiy) [inline main.go:138]: No error handling

File: main.go
Comment (kovetskiy) [inline main.go:139]: It doesn't look right that you use map[string]interface{} and then do a runtime assertion of a field execution. You should create a type like `ResponseExecution` and unmarshal the JSON into this struct. It's much safer than unmarshalling into a map[].

File: main.go
Comment (kovetskiy) [inline main.go:172]: Same problems as in previous methods here.

**Review by kovetskiy — CHANGES_REQUESTED** [review]
Hello there, I'm in kind of vacation right now, but here is the part 1 of the code review you've asked.

**Pull request closed without merging**
